program-runtime/src/compute_budget.rs

```diff
@@ -547,4 +550,142 @@ mod tests {
             ComputeBudget::default()
         );
     }
+
+    #[test]
+    fn test_process_instructions_disable_request_heap_frame() {
```

Comment:
These test cases are hard to understand and I have to guess a lot. Can you add more comments what exactly they test for?

Reply:
+1 please annotate conditions under test

Reply:
fair, tests are. refreshed and annotations added.